commands/template_pull_stack.go

```diff
@@ -0,0 +1,100 @@
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/openfaas/faas-cli/stack"
+	"github.com/spf13/cobra"
+)
+
+var (
+	templateURL string
+	customName  string
+)
+
+func init() {
+	templatePullStackCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
+	templatePullStackCmd.Flags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
+	templatePullStackCmd.PersistentFlags().StringVarP(&customName, "name", "n", "", "The custom name of the template")
+
+	templatePullCmd.AddCommand(templatePullStackCmd)
+}
+
+var templatePullStackCmd = &cobra.Command{
+	Use:   `stack`,
+	Short: `Downloads templates specified in the function definition yaml file`,
+	Long: `Downloads templates specified in the function yaml file, in the current directory
+	`,
+	Example: `
+  faas-cli template pull stack
+  faas-cli template pull stack -f myfunction.yml
+`,
+	RunE: runTemplatePullStack,
+}
+
+func runTemplatePullStack(cmd *cobra.Command, args []string) error {
+	templatesConfig, err := loadTemplateConfig()
+	if err != nil {
+		return err
+	}
+	err = pullTemplatesFromConfig(templatesConfig)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadTemplateConfig() (map[string]string, error) {
+	stackConfig, err := readStackConfig()
+	if err != nil {
+		return nil, err
+	}
+	return stackConfig.StackConfig.TemlpatesConfigs, nil
+}
+
+func readStackConfig() (stack.Configuration, error) {
+	configField := stack.Configuration{}
+
+	configFieldBytes, err := ioutil.ReadFile(yamlFile)
+	if err != nil {
+		return configField, fmt.Errorf("Error while reading files %s", err.Error())
+	}
+	unmarshallErr := yaml.Unmarshal(configFieldBytes, &configField)
+	if unmarshallErr != nil {
+		return configField, fmt.Errorf("Error while reading configuration: %s", err.Error())
+	}
+	if reflect.DeepEqual(configField, stack.Configuration{}) {
```

Comment:
I think it would be enough to do `len(configField.StackConfig.TemlpatesConfigs)==0`  This will be zero if the value is nil or empty and would avoid the use of reflection.

Reply:
Wow, :smile: I totally forgot about that feature, (the default 0), will do :+1:

Reply:
Done :heavy_check_mark: